apps/aechannel/src/aesc_utils.erl

```diff
@@ -479,8 +479,8 @@ process_force_progress(Tx, Addresses,
     PoITrees0 = trees_from_poi(Addresses, PoI),
     Reserve = aesc_channels:channel_reserve(Channel),
     PoITrees =
-        try aesc_offchain_update:apply_on_trees(Update, PoITrees0, NextRound,
-                                                      Reserve)
+        try aesc_offchain_update:apply_on_trees(Update, PoITrees0, Trees, Env,
```

Comment:
Is `Amount` (as in `{?OP_CALL_CONTRACT, _, _, _, Amount, _, _, GasPrice, Gas}`) used at all? When is it charged?

Reply:
I see the amount that is in the off-chain update:
* In happy path, affects the state hash meant to be compared. It makes sense.
* In happy path, affects the balance of initiator / responder that ends up in updated state channel. It makes sense.
* In case of exceptional handling / revert in VM call (not in the `catch _:_` in this file), it keeps affecting the balance of initiator / responder that ends up in updated state channel. This is in line with [contract call transaction](https://github.com/aeternity/protocol/blame/1c5b17b44da6ecd661bac0fe86c2b1c935f99e8d/contracts/contract_transactions.md#L111). (I added [comment in related ticket](https://www.pivotaltracker.com/story/show/160980601/comments/194950662).)

When is the `try ... catch _:_ ...` - with the cath-all - meant to kick in? In case of code implementation bug i.e. unexpected situation? I would expect in that case this function to terminate abnormally so to prevent bug to silently slip under consensus. Comments?

Reply:
The try-catch is for force-specific crashes. For example [when a participant is trying to spend more than one has](https://github.com/aeternity/epoch/blob/master/apps/aechannel/src/aesc_offchain_update.erl#L26). Currently - the only example, so probably a specific catch is better

Reply:
Made it thighter in [63cc334](https://github.com/aeternity/epoch/pull/1592/commits/63cc334feecb7241a237d33ffde8fb558b79aa79)

Reply:
> For example when a participant is trying to spend more than one has.

I see only in `check_min_amt`. And I see that check on the channel reserve is not enforced in the force progress tx check callback, that means that tx that violates reserve would be considered valid hence included in block and all gas limit taken from sender. This seems to make sense as a way of punishing user that produces attempt at illegal retrieval from channel reserve knowingly (as sender can specifies next round) though I recommend this to be clearly documented.